controller/src/beerocks/master/son_management.cpp

```diff
@@ -387,43 +387,54 @@ void son_management::handle_cli_message(Socket *sd,
         auto agent_mac         = database.get_node_parent_ire(hostap_mac);
         LOG(DEBUG) << "CLI beacon request for " << client_mac << " to " << hostap_mac;
 
-        auto request = message_com::create_vs_message<
-            beerocks_message::cACTION_CONTROL_CLIENT_BEACON_11K_REQUEST>(cmdu_tx);
-        if (request == nullptr) {
-            LOG(ERROR) << "Failed building ACTION_CONTROL_CLIENT_BEACON_11K_REQUEST message!";
+        auto beacon_metrics_request =
+            cmdu_tx.create(0, ieee1905_1::eMessageType::BEACON_METRICS_QUERY_MESSAGE);
+        if (!beacon_metrics_request) {
+            LOG(ERROR) << "Failed create BEACON_METRICS_QUERY_MESSAGE";
+            isOK = false;
+            break;
+        }
+
+        auto beacon_metrics_request_tlv = cmdu_tx.addClass<wfa_map::tlvBeaconMetricsQuery>();
+
+        if (!beacon_metrics_request_tlv) {
+            LOG(ERROR) << "Failed create BEACON_METRICS_QUERY_MESSAGE";
+            isOK = false;
+            break;
+        }
+        beacon_metrics_request_tlv->associated_sta_mac() = cli_request->client_mac();
+
+        beacon_metrics_request_tlv->operating_class()        = cli_request->op_class();
+        beacon_metrics_request_tlv->channel_number()         = cli_request->channel();
+        beacon_metrics_request_tlv->bssid()                  = cli_request->bssid();
+        beacon_metrics_request_tlv->reporting_detail_value() = 1;
+        beacon_metrics_request_tlv->ssid_length() = beerocks::message::WIFI_SSID_MAX_LENGTH;
+
+        if (!beacon_metrics_request_tlv->alloc_ssid(beerocks::message::WIFI_SSID_MAX_LENGTH)) {
+            LOG(ERROR) << "Couldn't allocate memory for SSID.";
             isOK = false;
             break;
         }
```

Comment:
```suggestion
        beacon_metrics_request_tlv->ssid_length() = message::WIFI_SSID_MAX_LENGTH;

        if (!beacon_metrics_request_tlv->alloc_ssid(message::WIFI_SSID_MAX_LENGTH)) {
            LOG(ERROR) << "Couldn't allocate memory for SSID.";
            isOK = false;
            break;
        }
```
the file contains `using namespace beerocks` so no need to explicitly specify it.